vita3k/mem/src/mem.cpp

```diff
@@ -202,23 +215,13 @@ Address alloc(MemState &state, size_t size, const char *name, unsigned int align
     return align_addr;
 }
 
-static void align_to_page(MemState &state, ProtectSegmentInfo &protect) {
-    const Address end = align(protect.addr + protect.size, state.page_size);
-    const Address addr = align_down(protect.addr, state.page_size);
-    const size_t size = end - addr;
-    protect.addr = addr;
-    protect.size = size;
+static void align_to_page(MemState &state, Address &addr, Address &size) {
```

Comment:
It's a bit confusing that addr and size have same type, it looks like size is also a guest pointer. (My suggestion is Address &addr, uint32_t &size)

Reply:
Indeed, you are right